src/python/WMCore/MicroService/Unified/MSRuleCleaner.py

```diff
@@ -50,27 +116,341 @@ def execute(self, reqStatus):
+
+        # Report the counters:
+        for pline in self.cleanuplines:
+            msg = "Workflows cleaned by pipeline: %s: %d"
+            self.logger.info(msg, pline.name, self.wfCounters['cleaned'][pline.name])
+        archived_num_requests = self.wfCounters['archived']
+        msg = "Workflows archived: %d"
+        self.logger.info(msg, self.wfCounters['archived'])
+        return total_num_requests, clean_num_requests, archived_num_requests
+
+    def _dispatchWflow(self, wflow, *args, **kwargs):
+        """
+        A function intended to dispatch a workflow (e.g based on its status)
+        through one or more functional pipelines in case there is some more
+        complicated logic involved in the order we execute them but not just
+        a sequentially
+        """
+        msg = "Called with wflow: %s"
+        self.logger.debug(msg, pformat(wflow))
+        # NOTE: The following dispatch logic is a subject to be changed at any time
+
+        # Do not clean any Resubmission, but still let them be archived
+        if wflow['RequestType'] == 'Resubmission':
+            wflow['ForceArchive'] = True
+            msg = "Skipping cleanup step for workflow: %s - RequestType is %s."
+            msg += "\nWill try to archive it directly."
+            self.logger.info(msg, wflow['RequestName'], wflow['RequestType'])
+        # Clean:
+        elif wflow['RequestStatus'] in ['rejected', 'aborted-completed']:
+            # NOTE: We do not check the ParentageResolved flag for these
+            #       workflows, but we do need to clean output data placement
+            #       rules from the agents for them
+            for pline in self.agentlines:
+                try:
+                    pline.run(wflow)
+                except Exception as ex:
+                    msg = "%s: General error from pipeline. Error: \n%s. "
+                    msg += "\nWill retry again in the next cycle."
+                    self.logger.exception(msg, pline.name, str(ex))
+                    continue
+                if wflow['CleanupStatus'][pline.name]:
+                    self.wfCounters['cleaned'][pline.name] += 1
+        elif wflow['RequestStatus'] == 'announced' and not wflow['ParentageResolved']:
+            # NOTE: We skip workflows which are not having 'ParentageResolved'
+            #       flag, but we still need some proper logging for these.
+            msg = "Skipping workflow: %s - 'ParentageResolved' flag set to false."
+            msg += "\nWill retry again in the next cycle."
+            self.logger.info(msg, wflow['RequestName'])
+        elif wflow['RequestStatus'] == 'announced' and wflow['ParentageResolved']:
+            for pline in self.cleanuplines:
+                try:
+                    pline.run(wflow)
+                except Exception as ex:
+                    msg = "%s: General error from pipeline. Error: \n%s. "
+                    msg += "\nWill retry again in the next cycle."
```

Comment:
Same comment here. Having the workflow in the error message is important too.

Reply:
Workflow name was  added to the message.